src/builder/command.rs

```diff
@@ -3740,6 +3740,12 @@ impl Command {
         Ok(matcher.into_inner())
     }
 
+    /// Prepare for completions by setting flags useful for that case.
+    /// Call this on the top-level [`Command`] before calling [`Command::build`].
+    pub fn prepare_build_for_completion(&mut self) {
+        self.g_settings.set(AppSettings::ExpandHelpSubcommandTrees);
+    }
```

Comment:
Should we just implicitly set this as part of `build()`?

Reply:
If it is desirable to have the subcommand trees present during `mangen` also, then sure.